src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/SymbolType.cs

```diff
@@ -286,7 +286,7 @@ public override Type MakeArrayType(int rank)
         public override int GetArrayRank()
         {
             if (!IsArray)
-                throw new NotSupportedException(SR.NotSupported_SubclassOverride);
+                throw new ArgumentException(SR.Argument_HasToBeArrayClass);
```

Comment:
InvalidOperationException instead?

Reply:
Followed exception type and message for similar cases
https://github.com/dotnet/runtime/blob/9f6ac8c5a23fde39dbcbd7f85dcd73a7da6c191c/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeModuleBuilder.cs#L1142-L1145